winch/codegen/src/masm.rs

```diff
@@ -448,6 +456,17 @@ pub(crate) trait MacroAssembler {
     /// This function will potentially emit a series of instructions.
     fn cmp_with_set(&mut self, src: RegImm, dst: Reg, kind: CmpKind, size: OperandSize);
 
+    /// Compare floats in src1 and src2 and put the result in dst.
+    /// In x86, this will emit multiple instructions.
+    fn float_cmp_with_set(
+        &mut self,
+        src1: Reg,
+        src2: Reg,
+        dst: Reg,
+        kind: CmpKind,
```

Comment:
I don't like this accepting `CmpKind` either since the method will panic if it's invoked with one of several variants of the `CmpKind` enum. I could reintroduce a `FloatCmpKind` and rename the existing `CmpKind` to `IntCmpKind`.

Reply:
I'm not too bothered with the panics, but as mentioned before, I think that shoehorning future use-cases into `CmpKind` will get tricky (e.g. vector operands), so introducing `FloatCmpKind` and renaming CmpKind to `IntCmpKind` now seems better.